### PR TITLE
Tools: add completion for subtest

### PR DESCRIPTION
**Pull request by khancyr**
Allow to complet subtest on autotest on zsh shell

Correct a function sim_vehicle.py for bash

### CODE REVIEW

File: Tools/autotest/autotest.py [hunk 1]
```diff
@@ -609,17 +609,37 @@ def run_tests(steps):
...
         subtests = tester.tests()
         print("%s:" % vehicle)
-        for subtest in sorted(subtests, key=lambda x : x[0]):
+        for subtest in sorted(subtests, key=lambda x: x[0]):
             (name, description, function) = subtest
             print("    %s: %s" % (name, description))
         print("")
 
+
+def list_subtests_v(vehicle_type):
```
Comment (peterbarker) [inline Tools/autotest/autotest.py:629]: Bytes are cheap here - could have spelt this out as `list_subtests_for_vehicle`

Comment (khancyr) [reply Tools/autotest/autotest.py:629]: sure!

**Comment by khancyr** [comment]
rebase and updated with @peterbarker comment

**Comment by peterbarker** [comment]
Merged, thanks!

Want to see the bash equivalent, 'though :-)

**Comment by khancyr** [comment]
bash version will come, but I need some time get back sanity before going back into completion stuff !

**Pull request merged**